src/software/util/time/duration.cpp

```diff
@@ -59,3 +61,10 @@ Duration Duration::operator-(const Duration &duration) const
 {
     return Duration::fromSeconds(getSeconds() - duration.getSeconds());
 }
+
+std::ostream &operator<<(std::ostream &output_stream, const Duration &duration)
+{
+    output_stream << std::setprecision(2) << std::fixed << duration.getSeconds() << "s";
+
+    return output_stream;
+}
```

Comment:
nit; we're missing the newline at the end of this file (and github doesn't like that), just adding a empty line after this one should fix it.